src/Storages/buildQueryTreeForShard.cpp

```diff
@@ -427,7 +490,8 @@ QueryTreeNodePtr buildQueryTreeForShard(const PlannerContextPtr & planner_contex
                 temporary_table_expression_node = executeSubqueryNode(
                     subquery_to_execute,
                     planner_context->getMutableQueryContext(),
-                    global_in_or_join_node.subquery_depth);
+                    global_in_or_join_node.subquery_depth,
+                    true);
```

Comment:
I'd avoid adding a new parameter and recommend setting `isDistinct` before calling `executeSubqueryNode`.